[macOS Ventura] Disable code signing in TestWebKitAPI and WebKitTestRunner

emw-apple opened the pull request:
#### 4febba7c264b9476b65518b1f492ca5c7726b73c
<pre>
Fix xcconfig and reland &quot;[macOS Ventura] Disable code signing in TestWebKitAPI and WebKitTestRunner&quot;
<a href="https://bugs.webkit.org/show_bug.cgi?id=242730">https://bugs.webkit.org/show_bug.cgi?id=242730</a>

Reviewed by NOBODY (OOPS!).

Variable expansions were missing a layer of parentheses.

* Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig:
* Tools/WebKitTestRunner/Configurations/WebKitTestRunner.xcconfig:
</pre>

aproskuryakov commented on Tools/WebKitTestRunner/Configurations/WebKitTestRunner.xcconfig:
```diff
@@ -45,12 +45,5 @@ INCLUDED_SOURCE_FILE_NAMES = mac/main.mm;
 
 CODE_SIGN_ENTITLEMENTS[sdk=macosx*] = Configurations/WebKitTestRunner.entitlements;
 
-CODE_SIGN_IDENTITY[sdk=macosx*] = $(CODE_SIGN_IDENTITY_$(CONFIGURATION));
```
I'm slightly curious about why TestWebKitAPI needs CODE_SIGN_IDENTITY in two xcconfigs, but WebKitTestRunner only needs it in DebugRelease.

emw-apple replied:
WebKitTestRunner has these same CODE_SIGN_IDENTITY declarations already: https://github.com/WebKit/WebKit/blob/cb8e510d4f433626e5fe6f9ab6333e46f8b337b6/Tools/WebKitTestRunner/Configurations/DebugRelease.xcconfig#L33-L36

So this change aligns TestWebKitAPI's signing identity behavior with WKTR's.

emw-apple replied:
Actually, I was misreading, you were right to point out the discontinuity.

Updated both projects to set ad-hoc codesigning unconditionally outside of DebugRelease. This is other production-buildable tools (DumpRenderTree, MiniBrowser, MobileMiniBrowser) do.

aproskuryakov reviewed (approved):
I think that "-" is called ad hoc, not null. Also, I'm not sure if there is any "default" identity, so the commit message seems a little misleading.

JonWBedard reviewed (approved):
r+, although we really should wait to make sure EWS sim can run tests before landing this.

emw-apple committed:
[macOS Ventura] Disable code signing in TestWebKitAPI and WebKitTestRunner
https://bugs.webkit.org/show_bug.cgi?id=242730

Reviewed by Jonathan Bedard and Alexey Proskuryakov.

Xcode 14 no longer accepts an empty signing identity on macOS, and is
now trying to sign these tools with an identity it cannot find.
Previously, they were completely unsigned.

Set CODE_SIGNING_ALLOWED=NO to bypass signing and retain the status quo.

* Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig:
* Tools/WebKitTestRunner/Configurations/WebKitTestRunner.xcconfig:

Canonical link: https://commits.webkit.org/253391@main
